src/QueryBuilder/typeQueries/referenceQuery.test.ts

```diff
@@ -11,68 +11,145 @@ const fhirSearchParametersRegistry = new FHIRSearchParametersRegistry('4.0.1');
-                  "Organization/organizationId",
-                ],
-              },
-            }
-        `);
+    describe('searching with just {id}', () => {
+        test('one target type found', () => {
+            expect(
+                referenceQuery(organizationParam, 'organizationId', true, 'https://base-url.com', 'organization', [
+                    'Organization',
+                ]),
+            ).toMatchInlineSnapshot(`
+              Object {
+                "terms": Object {
+                  "managingOrganization.reference.keyword": Array [
+                    "https://base-url.com/Organization/organizationId",
+                    "Organization/organizationId",
+                  ],
+                },
+              }
+            `);
+        });
+        test('many target types found', () => {
+            expect(
+                referenceQuery(organizationParam, 'organizationId', true, 'https://base-url.com', 'organization', [
+                    'Organization',
+                    'Group',
+                ]),
+            ).toMatchInlineSnapshot(`
+              Object {
+                "terms": Object {
+                  "managingOrganization.reference.keyword": Array [
+                    "https://base-url.com/Organization/organizationId",
+                    "Organization/organizationId",
+                    "https://base-url.com/Group/organizationId",
+                    "Group/organizationId",
+                  ],
+                },
+              }
+            `);
+        });
+        test('no target types found', () => {
+            expect(() =>
+                referenceQuery(organizationParam, 'organizationId', false, 'https://base-url.com', 'organization'),
+            ).toThrow(InvalidSearchParameterError);
+        });
     });
-    test('just id search, many types found', () => {
-        expect(referenceQuery(organizationParam, 'organizationId', true, 'organization', ['Organization', 'Group']))
-            .toMatchInlineSnapshot(`
-            Object {
-              "terms": Object {
-                "managingOrganization.reference.keyword": Array [
-                  "Organization/organizationId",
-                  "Group/organizationId",
-                ],
-              },
-            }
-    `);
+    test('invalid modifier', () => {
+        expect(() =>
+            referenceQuery(
+                organizationParam,
+                'organizationId',
+                false,
+                'https://base-url.com',
+                'organization',
+                [],
+                'exact',
+            ),
+        ).toThrow(InvalidSearchParameterError);
     });
-    test('just id search, no types found', () => {
-        expect(() => referenceQuery(organizationParam, 'organizationId', false, 'organization')).toThrow(
-            InvalidSearchParameterError,
-        );
+    test('search does not match expect id or url', () => {
```

Comment:
nit: I think the test name could be a bit more descriptive
```suggestion
    test('search value is not an URL nor has the format <resourceType>/<id>', () => {
```